client/keys.go

```diff
@@ -91,6 +90,7 @@ func (key *PrivateKey) execute(op gokeyless.Op, msg []byte) ([]byte, error) {
 		SNI:      key.sni,
 	})
 	if err != nil {
+		defer conn.Close()
```

Comment:
yes it can be closed immediately but I just copy it from below. And yes the conn is kept open when no error has occurred. For each connection, we have a separate thread which keeps using the connection for ping. If the connection goes bad, it is closed.